catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/BroadcastBricksTest.java

```diff
@@ -0,0 +1,143 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import android.widget.EditText;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.BroadcastScript;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.bricks.BroadcastBrick;
+import org.catrobat.catroid.content.bricks.BroadcastWaitBrick;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+
+@RunWith(AndroidJUnit4.class)
+public class BroadcastBricksTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(this.getClass().getSimpleName());
+		baseActivityTestRule.launchActivity();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		TestUtils.deleteProjects(this.getClass().getSimpleName());
+	}
+
+	@Test
+	public void testAllBroadcastBricksSpinnersShowTheNewAddedMessage() {
+		String message = "Oida!";
+		int firstBrickPosition = 0;
+
+		createNewMessageOnSpinner(firstBrickPosition, message);
+		onBrickAtPosition(firstBrickPosition)
+				.onSpinner(R.id.brick_broadcast_spinner)
+				.checkShowsText(message);
+
+		checkBricksSpinnerHasAndShowsText(1, message);
+		checkBricksSpinnerHasAndShowsText(2, message);
+		checkBricksSpinnerHasAndShowsText(3, message);
+		checkBricksSpinnerHasAndShowsText(4, message);
+		checkBricksSpinnerHasAndShowsText(5, message);
+	}
+
+	private void createNewMessageOnSpinner(int position, String message) {
```

Comment:
theres already 3 occurences of pretty much exactly this method in BroadcastAndWaitBrickTest, BroadcastReceiverTest, and BroadcastSendBrickTest.
this would be a nice time to literally wrap this up (in a eg. BroadcastSpinnerWrapper)

Reply:
done